src/Symfony/Component/Uid/Tests/ArgumentResolver/UidValueResolverTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uid\Tests\ArgumentResolver;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Uid\AbstractUid;
+use Symfony\Component\Uid\ArgumentResolver\UidValueResolver;
+use Symfony\Component\Uid\Factory\UlidFactory;
+use Symfony\Component\Uid\Ulid;
+use Symfony\Component\Uid\UuidV1;
+use Symfony\Component\Uid\UuidV4;
+
+class UidValueResolverTest extends TestCase
+{
+    /**
+     * @dataProvider provideSupports
+     */
+    public function testSupports(bool $expected, Request $request, ArgumentMetadata $argument)
+    {
+        $this->assertSame($expected, (new UidValueResolver())->supports($request, $argument));
+    }
+
+    public function provideSupports()
+    {
+        return [
+            // Variadic argument
+            [false, new Request([], [], ['foo' => (string) $uuidV4 = new UuidV4()]), new ArgumentMetadata('foo', UuidV4::class, true, false, null)],
```

Comment:
use this comment as the test case name?

```suggestion
           'Variadic argument' => [false, new Request([], [], ['foo' => (string) $uuidV4 = new UuidV4()]), new ArgumentMetadata('foo', UuidV4::class, true, false, null)],
```